app/(auth)/signup/_components/signupForm.tsx

```diff
@@ -0,0 +1,109 @@
+'use client';
+
+import { Controller, FieldValues, useForm } from 'react-hook-form';
+
+import { AxiosError } from 'axios';
+import { useRouter } from 'next/navigation';
+
+import Button from '@/components/button/button';
+import Input from '@/components/input/input';
+import { postSignUpInfo } from '@/services/api';
+
+import MemberType from './memberType';
+
+const EMAIL_REGEX = /^[A-Za-z0-9]([-_.]?[A-Za-z0-9])*@[a-z0-9]([-_.]?[A-Za-z0-9])*\.[A-Za-z]{2,3}$/;
+const PASSWORD_REGEX = /^.{8,}$/;
+
+export default function SignUpForm() {
+  const router = useRouter();
+
+  const {
+    register,
+    control,
+    handleSubmit,
+    watch,
+    formState: { errors },
+  } = useForm<FieldValues>({ mode: 'all', defaultValues: { type: 'employee' } });
+
+  const FORM_FIELDS = [
+    {
+      name: 'email',
+      label: '이메일',
+      type: 'text',
+      placeholder: '이메일을 입력해주세요.',
+      validation: {
+        required: '이메일을 입력해주세요.',
+        pattern: {
+          value: EMAIL_REGEX,
+          message: '올바른 이메일 주소가 아닙니다.',
+        },
+      },
+    },
+    {
+      name: 'password',
+      label: '비밀번호',
+      type: 'password',
+      placeholder: '비밀번호를 입력해주세요.',
+      validation: {
+        required: '비밀번호를 입력해주세요.',
+        pattern: {
+          value: PASSWORD_REGEX,
+          message: '비밀번호를 8자 이상 입력해주세요.',
+        },
+      },
+    },
+    {
+      name: 'password_repeat',
+      label: '비밀번호 확인',
+      type: 'password',
+      placeholder: '비밀번호를 한 번 더 입력해주세요.',
+      validation: {
+        validate: (value: string) => value === watch('password') || '비밀번호가 일치하지 않습니다.',
```

Comment:
매개변수명을 여기서는 더 명확하게 잡아주면 좋을 것 같구요~
react-hook-form에서 watch는 특정 form field의 값을 계속해서 관찰하겠다는 용도로 사용되는 함수이기 때문에 이벤트가 발생했을 때 한번만 값을 가져와도 되는 경우라면 getValues 함수를 사용해주는게 더 좋을 것 같아요!
```suggestion
        validate: (confirmPassword: string) => value === getValues('password') || '비밀번호가 일치하지 않습니다.',
```